File: Python/Math-Array-String/32.py
# 2154. Keep Multiplying Found Values by Two


from typing import List

class Solution:
    def findFinalValue(self, nums: List[int], original: int) -> int:
        # Convert nums to a set for O(1) lookup time
        nums_set = set(nums)
        # Keep checking and multiplying original as long as it's found in nums_set
        while original in nums_set:
            original *= 2
        
        return original


sol = Solution()
output = sol.findFinalValue([5,3,6,1,12], 3)
print(output)

File: JavaScript/Math-Array-String/32.js
// 2154. Keep Multiplying Found Values by Two

/**
 * @param {number[]} nums
 * @param {number} original
 * @return {number}
 */
var findFinalValue = function(nums, original) {
    // Convert nums to a Set for faster lookup
    let numsSet = new Set(nums);

    // Keep checking and multiplying original as long as it's found in numsSet
    while (numsSet.has(original)) {
        original *= 2;
    }

    return original;
};


console.log(findFinalValue([5,3,6,1,12], 3));
